python/rmm/_cuda/gpu.py

```diff
@@ -1,59 +1,43 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 
-from rmm._cuda.gpu cimport (
-    CUresult,
-    cudaDeviceAttr,
-    cudaDeviceGetAttribute,
-    cudaDeviceProp,
-    cudaDriverGetVersion,
-    cudaError,
-    cudaError_t,
-    cudaGetDeviceCount,
-    cudaGetDeviceProperties,
-    cudaGetErrorName,
-    cudaGetErrorString,
-    cudaRuntimeGetVersion,
-    cuDeviceGetName,
-    cuGetErrorName,
-    cuGetErrorString,
-)
-
-from enum import IntEnum
+from cuda import cuda, cudart
 
 
 class CUDARuntimeError(RuntimeError):
-
-    def __init__(self, cudaError_t status):
+    def __init__(self, status: cudart.cudaError_t):
         self.status = status
-        cdef str name = cudaGetErrorName(status).decode()
-        cdef str msg = cudaGetErrorString(status).decode()
+
+        err, name = cudart.cudaGetErrorName(status)
+        if err != cudart.cudaError_t.cudaSuccess:
+            raise CUDARuntimeError(err.value)
+
+        err, msg = cudart.cudaGetErrorString(status)
+        if err != cudart.cudaError_t.cudaSuccess:
+            raise CUDARuntimeError(err.value)
+
         super(CUDARuntimeError, self).__init__(
-            '%s: %s' % (name, msg))
+            "%s: %s" % (name.decode(), msg.decode())
```

Comment:
Let's use f-strings for conciseness and readability. 🤠
```suggestion
            f"{name.decode()}: {msg.decode()}"
```